Client/game_sa/CDynamicPool.h

```diff
@@ -0,0 +1,192 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        game_sa/CDynamicPool.h
+ *  PURPOSE:     Custom implementation for SA pools
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#pragma once
+
+#include <vector>
+#include <array>
+#include <memory>
+
+template <typename PoolObjT>
+class CDynamicPoolPart
+{
+public:
+    explicit CDynamicPoolPart(std::size_t size) : m_size{size}, m_usedSize{0}, m_lastFreeSlot{0}
+    {
+        m_items = std::make_unique<PoolObjT[]>(size);
+        m_usedSlots = std::make_unique<bool[]>(size);
```

Comment:
Did you think about using a stack of free indices? It will be much more efficient than iterating over array of bools searching for a free item.

Reply:
I like your idea. It's less memory efficient but gives better performance. 
My current solution takes 810 KB for 90k objects. 9 byte/object. The game engine has the same memory usage.
Your solution requires 1080 KB for the same amount of objects (12 byte/object). 
If you think that extra memory usage isn't significant, i can use a stack here

Reply:
Personally, I don't see these extra ~200 KB as a problem. The memory budgets in SA were set taking in account the possibilities of PS2, but most of todays PCs have much more RAM and we can allow ourselves to use it more freely. A stack can be implemented on preallocated(because we know the size of it in advance) `std::vector`(just `push_back` and `pop_back`) as it will be for efficient than `std::stack` because the latter leads to excessive memory fragmentation(especially on **MSVC**).

Reply:
Done